### PR TITLE
Add ray tracing shader descriptor indexing validation

### PR DESCRIPTION
**Pull request by jmacnak-nv**
Updates gpu validation to support descriptor indexing shader checks for
ray tracing ray generation, intersection, any hit, closest hit, miss, and callable shaders.

Note this PR depends on https://github.com/KhronosGroup/SPIRV-Tools/pull/2733 - just want to start review process to get feedback.

### CODE REVIEW

**Comment by jmacnak-nv** [comment]
(re-based now that dependencies have merged)

**Comment by Tony-LunarG** [comment]
32 bit compiler warnings:
C:\j\w8\32-bit\Release\tests\vklayertests_descriptor_renderpass_framebuffer.cpp(482): warning C4244: 'argument': conversion from 'VkDeviceSize' to 'std::size_t', possible loss of data [C:\j\w8\32-bit\Release\build32\tests\vk_layer_validation_tests.vcxproj]

C:\j\w8\32-bit\Release\tests\vklayertests_descriptor_renderpass_framebuffer.cpp(535): warning C4244: 'argument': conversion from 'VkDeviceSize' to 'std::size_t', possible loss of data [C:\j\w8\32-bit\Release\build32\tests\vk_layer_validation_tests.vcxproj]

C:\j\w8\32-bit\Release\tests\vklayertests_descriptor_renderpass_framebuffer.cpp(1155): warning C4244: 'argument': conversion from 'VkDeviceSize' to 'unsigned int', possible loss of data [C:\j\w8\32-bit\Release\build32\tests\vk_layer_validation_tests.vcxproj]

C:\j\w8\32-bit\Release\tests\vklayertests_descriptor_renderpass_framebuffer.cpp(1156): warning C4244: 'argument': conversion from 'VkDeviceSize' to 'std::size_t', possible loss of data [C:\j\w8\32-bit\Release\build32\tests\vk_layer_validation_tests.vcxproj]

**Comment by Tony-LunarG** [comment]
Could you trim the commit messages that travis is complaining about?  I think that and the warnings are all that is needed.

**Comment by jmacnak-nv** [comment]
Hmm, are you looking at an older snapshot? I believe the latest squashed commits have the appropriate static_cast's in place for those warnings.

**Comment by Tony-LunarG** [comment]
Could be - I'll try again

**Comment by jmacnak-nv** [comment]
Updated commit messages. Let me know if any of those abbreviations sound weird.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -5145,50 +5146,72 @@ bool ValidationStateTracker::PreCallValidateCreateRayTracingPipelinesNV(VkDevice
...
-        pipe_state->push_back(std::unique_ptr<PIPELINE_STATE>(new PIPELINE_STATE));
-        (*pipe_state)[i]->initRayTracingPipelineNV(this, &pCreateInfos[i]);
-        (*pipe_state)[i]->pipeline_layout = *GetPipelineLayout(pCreateInfos[i].layout);
+        // Create and initialize internal tracking data structure
+        crtpl_state->pipe_state.push_back(unique_ptr<PIPELINE_STATE>(new PIPELINE_STATE));
+        crtpl_state->pipe_state.back()->initRayTracingPipelineNV(this, &pCreateInfos[i]);
+        crtpl_state->pipe_state.back()->pipeline_layout = *GetPipelineLayout(pCreateInfos[i].layout);
     }
-    return false;
+    return skip;
```
[the remark this answers is not in the document]
Comment (jmacnak-nv) [reply layers/core_validation.cpp]: Nope, removed here and also in PreCallValidateCreateComputePipelines.

**Comment by jmacnak-nv** [comment]
> BTW - we had to update the driver on our CI system because the new test was getting an access violation in the graphics driver (416.94), but updating to a newer driver fixed the problem. FYI

Can you share more details about the system that CI is running on (CPU, OS, graphics card)?

**Comment by Tony-LunarG** [comment]
Intel Xeon E5-2650v3, Win 10 x64, GeForce RTX 2070

**Comment by Tony-LunarG** [comment]
@jmacnak-nv ready for merge?

**Comment by jmacnak-nv** [comment]
I believe we're ready.

**Pull request merged**